book-3-web-api/chapters/REACT_WITH_API.md

```diff
@@ -0,0 +1,210 @@
+
+  useEffect(() => {
+    getAllPosts();
+  }, []);
+
+  return (
+    <div>
+      {posts.map((post) => (
+        <div key={post.id}>
+          <img src={post.imageUrl} alt={post.title} />
+          <p>
+            <strong>{post.title}</strong>
+          </p>
+          <p>{post.caption}</p>
+        </div>
+      ))}
+    </div>
+  );
+};
+
+export default PostList;
+```
+
+Nothing too fancy here. When the component loads, it will call the `getAllPosts` method it recieves from the provider and render a list of posts.
+
+## Wiring It Up
+
+We have our nice provider and component so lets use them. Replace App.js with the following code
+
+```js
+import React from "react";
+import "./App.css";
+import { PostProvider } from "./providers/PostProvider";
+import PostList from "./components/PostList";
+
+function App() {
+  return (
+    <div className="App">
+      <PostProvider>
+        <PostList />
+      </PostProvider>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Make sure your API server is still running, and run `npm start` in your terminal.
+
+## Styling w/ Reactstrap
+
+Lets make our app look a little nicer. Install reactstrap by running the following command (Make sure to run this from the same directory your package.json file is in)
+
+```sh
+npm install --save bootstrap reactstrap
+```
+
+Now import the css file into your `index.js` file
+
+```js
+import 'bootstrap/dist/css/bootstrap.min.css';
+```
+
+#### Making a Post Component
+
+Our posts might get a little more complex so it's probably a good idea to separate it out into it's own component. Create a `Post.js` file in the components directory with the following code
+
+```js
+import React from "react";
+import { Card, CardImg, CardBody } from "reactstrap";
+
+const Post = ({ post }) => {
+  return (
+    <Card className="m-4">
+      <p className="text-left px-2">Posted by: {post.userProfile.name}</p>
+      <CardImg top src={post.imageUrl} alt={post.title} />
+      <CardBody>
+        <p>
+          <strong>{post.title}</strong>
+        </p>
+        <p>{post.caption}</p>
+      </CardBody>
+    </Card>
+  );
+};
+
+export default Post;
+```
+
+Again, nothing fancy here; we're just using the Card component that comes with reactstrap to organize some of the post details.
+
+Now lets update the `PostList` component to use the new `Post` component
+
+```js
+import React, { useContext, useEffect } from "react";
+import { PostContext } from "../providers/PostProvider";
+import Post from "./Post";
+
+const PostList = () => {
+  const { posts, getAllPosts } = useContext(PostContext);
+
+  useEffect(() => {
+    getAllPosts();
+  }, []);
+
+  return (
+    <div className="container">
+      <div className="row justify-content-center">
+        <div class="cards-column">
+          {posts.map((post) => (
+            <Post key={post.id} post={post} />
+          ))}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default PostList;
+```
+
+## Exercise
```

Comment:
I think we should add an exercise to have them list all the comments beneath each post. We can refactor after we do routing to have them only show comments on the Post "details" page, but I think another display-only component will help them get back into the swing of react.

I'm assuming we'll have a comments array on each post.